pkg/dataplane/deployment.go

```diff
@@ -496,3 +504,24 @@ func (d *Deployer) addServiceExtraMounts(
 
 	return d.AeeSpec, nil
 }
+
+func (d *Deployer) setAdditionalAEEOptions() error {
+	var test bool
+	cm, _, err := dataplaneutil.GetDataSourceCmSecret(d.Ctx, d.Helper, d.NodeSet.Namespace, d.Deployment.Spec.AnsibleEEConfig)
+
+	if err != nil || cm == nil {
+		d.AeeSpec.EnvConfigMapName = "openstack-aee-default-env"
+		return fmt.Errorf("aditional AEE options not found")
+	}
+
+	// Check presence of the key, type and contents
+	configMapName, test := cm.Data["envConfigMapName"]
```

Comment:
is it that the CM named d.Deployment.Spec.AnsibleEEConfig stores the name of another CM that actually stores the config? if so, why is that the case?

Reply:
Idea is that the `AnsibleEEConfig` will be catch all place for all the configuration options that don't necessarily need their own field. Whether they will be defined as config maps, such as in this case, or plain values.

In this case we are storing name of a config map, which is used for environment variables. So yes, we could just remove that one level and get the same value. But it would prevent us from using this field for anything else. At some point in the future, if customer came along and asked, we would have to add another field, and another. This way we just put it in a general config map, and let this function do whatever is necessary with it.